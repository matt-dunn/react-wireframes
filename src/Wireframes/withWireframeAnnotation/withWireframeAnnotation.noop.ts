/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

/* istanbul ignore file */
import { ComponentType } from "react";

export const withWireframeAnnotation = <P>(WrappedComponent: ComponentType<P> | string) => WrappedComponent;
