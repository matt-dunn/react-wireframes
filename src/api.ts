/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import { ComponentType, ReactNode } from "react";
import { remove } from "lodash";

type APIOptions = {
    updater: (components: WireFrameComponents) => void;
    highlightNote?: (component?: WireFrameComponent) => void;
}

export type WireFrameComponentOptions = {
    title: ReactNode;
    description: ReactNode;
}

export type WireFrameComponent = {
    id: number;
    Component: ComponentType<any>;
    count: number;
    options: WireFrameComponentOptions;
}

export type WireFrameComponents = WireFrameComponent[];

type OpenCallback = (isOpen: boolean) => void;

export type WireFrameAnnotationAPI = {
    setOptions: (options: APIOptions) => APIOptions;
    getComponents: () => WireFrameComponents;
    register: (Component: ComponentType<any>, options: WireFrameComponentOptions) => WireFrameComponent;
    unregister: (Component: ComponentType<any>) => void;
    highlightNote: (Component: ComponentType<any> | undefined) => void;
    setOpen: (isOpen: boolean) => boolean;
    onOpen: (cb: OpenCallback) => {unregister: () => void};
    isOpen: () => boolean;
}

export const API = function API(defaultOptions?: APIOptions): WireFrameAnnotationAPI {
  let components: WireFrameComponents = [];
  let apiOptions: APIOptions = defaultOptions || {} as APIOptions;

  let openCallbacks: OpenCallback[] = [];
  let isOpen = false;

  const getComponent = (Component: ComponentType<any> | undefined): WireFrameComponent | undefined => Component && components.find(c => c.Component === Component);

  return {
    setOptions: (options) => {
      apiOptions = options;
      return apiOptions;
    },
    getComponents: () => components,
    register: (Component, options) => {
      const component = getComponent(Component);

      if (component) {
        component.count += 1;

        if (apiOptions.updater) {
          apiOptions.updater(components);
        }

        return component;
      }

      const newComponent = {
        id: components.length + 1,
        Component,
        count: 1,
        options,
      };

      components = [...components, newComponent];

      if (apiOptions.updater) {
        apiOptions.updater(components);
      }

      return newComponent;
    },
    unregister: (Component) => {
      const component = getComponent(Component);

      if (component) {
        component.count -= 1;

        if (component.count === 0) {
          components = remove(components, c => c.Component !== Component);
        }
      }

      if (apiOptions.updater) {
        apiOptions.updater(components);
      }
    },
    highlightNote: Component => apiOptions && apiOptions.highlightNote && apiOptions.highlightNote(getComponent(Component)),
    setOpen: (open) => {
      isOpen = open;
      openCallbacks.forEach(cb => cb(isOpen));
      return isOpen;
    },
    onOpen: (cb) => {
      openCallbacks = [...openCallbacks, cb];
      return {
        unregister: () => {
          openCallbacks = remove(openCallbacks, c => c !== cb);
        },
      };
    },
    isOpen: () => isOpen,
  };
};
