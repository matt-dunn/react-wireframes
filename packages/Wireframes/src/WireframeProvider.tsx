/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React, {
  createContext, ReactNode,
} from "react";

import { WireframeAnnotation, WireframeAnnotationAPI } from "./api";

export const WireframeAnnotationContext = createContext<WireframeAnnotationAPI | undefined>(undefined);
export const WireframeAnnotationComponentContext = createContext<WireframeAnnotation | undefined>(undefined);

type WireframeProviderProps = {
  children: ReactNode;
  api: WireframeAnnotationAPI;
}

export const WireframeProvider = ({ children, api }: WireframeProviderProps) => (
  <WireframeAnnotationContext.Provider value={api}>
    {children}
  </WireframeAnnotationContext.Provider>
);
