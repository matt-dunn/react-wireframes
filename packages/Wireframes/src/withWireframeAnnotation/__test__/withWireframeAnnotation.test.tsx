/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React, { ComponentType, ReactElement } from "react";
import { mount, shallow } from "enzyme";
import { act } from "react-dom/test-utils";

import { WireframeProvider } from "../../WireframeProvider";
import { API, WireframeAnnotationAPI } from "../../api";

import { withWireframeAnnotation } from "../withWireframeAnnotation";

describe("Wireframe: withWireframeAnnotation", () => {
  let api: WireframeAnnotationAPI;
  let Fragment: ReactElement;
  let WrappedComponent: ComponentType<any>;
  let ComponentTree: ReactElement;
  let highlightNote: any;

  beforeEach(() => {
    highlightNote = jest.fn();

    api = API({
      highlightNote,
    } as any);

    WrappedComponent = withWireframeAnnotation({
      title: <div>Title</div>,
      description: <div>Description.</div>,
    })(() => <div>MOCK COMPONENT</div> as any);

    ComponentTree = <WrappedComponent />;

    Fragment = (
      <WireframeProvider api={api}>
        {ComponentTree}
      </WireframeProvider>
    );
  });

  it("should render without identifier when closed", () => {
    const wrapper = mount(Fragment);

    const annotation = api.getAnnotations()[0];

    expect(annotation).toEqual({
      id: 1,
      count: 1,
      Component: (WrappedComponent as any).Component,
      options: {
        title: <div>Title</div>,
        description: <div>Description.</div>,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("should render annotation identifier when open", () => {
    const wrapper = mount(Fragment);

    act(() => {
      api.setOpen(true);
    });

    expect(highlightNote).not.toHaveBeenCalled();

    expect(wrapper).toMatchSnapshot();
  });

  it("should highlight annotation", () => {
    const wrapper = mount(Fragment);

    act(() => {
      api.setOpen(true);
    });

    expect(highlightNote).not.toHaveBeenCalled();

    wrapper.simulate("mousemove");

    expect(highlightNote).toHaveBeenCalledWith(api.getAnnotations()[0]);

    wrapper.update();

    expect(wrapper).toMatchSnapshot();

    wrapper.simulate("mouseleave");

    expect(highlightNote).toHaveBeenCalledWith(undefined);
  });

  it("should show annotation highlighted", () => {
    const wrapper = mount(
      <WireframeProvider api={api}>
        <WrappedComponent isHighlighted />
      </WireframeProvider>,
    );

    act(() => {
      api.setOpen(true);
    });

    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });

  it("should nest providers", () => {
    const WrappedFragment = withWireframeAnnotation({
      title: <div>Title</div>,
      description: <div>Description.</div>,
    })(WireframeProvider);

    const wrapper = mount(
      <WireframeProvider api={api}>
        <WrappedFragment api={API()}>
          children
        </WrappedFragment>
      </WireframeProvider>,
    );

    act(() => {
      api.setOpen(true);
    });

    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });

  it("should unregister when unmounted", () => {
    const wrapper = mount(Fragment);

    expect(api.getAnnotations().length).toEqual(1);

    wrapper.unmount();
    expect(api.getAnnotations().length).toEqual(0);
  });

  it("should throw exception if api is not provided", () => {
    expect(() => {
      shallow(ComponentTree);
    }).toThrow(TypeError);
  });
});
