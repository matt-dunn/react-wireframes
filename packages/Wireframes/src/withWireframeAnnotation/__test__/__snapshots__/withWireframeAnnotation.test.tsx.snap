// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wireframe: withWireframeAnnotation should highlight annotation 1`] = `
.emotion-7 {
  position: relative;
}

.emotion-7 [disabled] {
  pointer-events: none;
}

.emotion-7.outlined,
.emotion-7.outline:hover {
  z-index: 5000;
}

.emotion-7.outlined > *,
.emotion-7.outline:hover > * {
  box-shadow: 0 0 0 1px #4086f7 !important;
}

.emotion-7.outlined > [data-annotation-identifier],
.emotion-7:hover > [data-annotation-identifier] {
  -webkit-transition: opacity 0ms,visibility 0ms;
  transition: opacity 0ms,visibility 0ms;
  opacity: 1;
}

.emotion-3 {
  position: absolute;
  top: -1em;
  left: -1em;
  z-index: 4000;
  cursor: default;
  -webkit-transition: opacity 250ms,visibility 250ms;
  transition: opacity 250ms,visibility 250ms;
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.enter {
  opacity: 0;
  visibility: hidden;
}

.emotion-3.fade-Identifier.enter-active {
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.exit {
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.exit-active {
  opacity: 0;
  visibility: hidden;
}

.emotion-1 {
  all: initial;
  box-sizing: border-box;
  line-height: 1;
  border-radius: 2em;
  color: #333;
  background-color: yellow;
  border-color: #caca00;
  white-space: nowrap;
  min-width: 2em;
  max-width: initial;
  min-height: auto;
  max-height: initial;
  height: 2em;
  padding: 0.25em 0.5em;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: initial;
  overflow: visible;
  font-style: normal;
  font-weight: normal;
  font-size: 1rem;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: absolute;
  top: -1em;
  left: -1em;
  z-index: 4000;
  cursor: default;
  -webkit-transition: opacity 250ms,visibility 250ms;
  transition: opacity 250ms,visibility 250ms;
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.enter {
  opacity: 0;
  visibility: hidden;
}

.emotion-1.fade-Identifier.enter-active {
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.exit {
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.exit-active {
  opacity: 0;
  visibility: hidden;
}

<WireframeProvider
  api={
    Object {
      "getAnnotations": [Function],
      "getParentReference": [Function],
      "highlightNote": [Function],
      "isOpen": [Function],
      "onOpen": [Function],
      "register": [Function],
      "setOpen": [Function],
      "setOptions": [Function],
      "setParentReference": [Function],
      "unregister": [Function],
    }
  }
>
  <WrappedWireframeAnnotation>
    <Wrapper
      className="outline"
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      show={true}
    >
      <span
        className="outline emotion-7 emotion-8"
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
      >
        <Identifier
          annotation={
            Object {
              "Component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "displayName": "withWireframeAnnotation(Component)",
                "type": [Function],
              },
              "count": 1,
              "id": 1,
              "options": Object {
                "description": <div>
                  Description.
                </div>,
                "title": <div>
                  Title
                </div>,
              },
            }
          }
          show={true}
        >
          <CSSTransition
            className="fade-Identifier"
            classNames=""
            in={true}
            mountOnEnter={true}
            timeout={250}
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="fade-Identifier"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={250}
              unmountOnExit={true}
            >
              <IdentifierContainer
                annotation={
                  Object {
                    "Component": Object {
                      "$$typeof": Symbol(react.memo),
                      "compare": null,
                      "displayName": "withWireframeAnnotation(Component)",
                      "type": [Function],
                    },
                    "count": 1,
                    "id": 1,
                    "options": Object {
                      "description": <div>
                        Description.
                      </div>,
                      "title": <div>
                        Title
                      </div>,
                    },
                  }
                }
                className="fade-Identifier"
                data-annotation-identifier={true}
              >
                <Memo(IdentifierBase)
                  annotation={
                    Object {
                      "Component": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "displayName": "withWireframeAnnotation(Component)",
                        "type": [Function],
                      },
                      "count": 1,
                      "id": 1,
                      "options": Object {
                        "description": <div>
                          Description.
                        </div>,
                        "title": <div>
                          Title
                        </div>,
                      },
                    }
                  }
                  className="fade-Identifier emotion-3 emotion-0"
                  data-annotation-identifier={true}
                >
                  <IdentifierContainer
                    className="fade-Identifier emotion-3 emotion-0"
                    data-annotation-identifier={true}
                  >
                    <cite
                      className="fade-Identifier emotion-0 emotion-1 emotion-2"
                      data-annotation-identifier={true}
                    >
                      1
                    </cite>
                  </IdentifierContainer>
                </Memo(IdentifierBase)>
              </IdentifierContainer>
            </Transition>
          </CSSTransition>
        </Identifier>
        <withWireframeAnnotation(Component)>
          <Component>
            <div>
              MOCK COMPONENT
            </div>
          </Component>
        </withWireframeAnnotation(Component)>
      </span>
    </Wrapper>
  </WrappedWireframeAnnotation>
</WireframeProvider>
`;

exports[`Wireframe: withWireframeAnnotation should nest providers 1`] = `
.emotion-7 {
  position: relative;
}

.emotion-7 [disabled] {
  pointer-events: none;
}

.emotion-7.outlined,
.emotion-7.outline:hover {
  z-index: 5000;
}

.emotion-7.outlined > *,
.emotion-7.outline:hover > * {
  box-shadow: 0 0 0 1px #4086f7 !important;
}

.emotion-7.outlined > [data-annotation-identifier],
.emotion-7:hover > [data-annotation-identifier] {
  -webkit-transition: opacity 0ms,visibility 0ms;
  transition: opacity 0ms,visibility 0ms;
  opacity: 1;
}

.emotion-3 {
  position: absolute;
  top: -1em;
  left: -1em;
  z-index: 4000;
  cursor: default;
  -webkit-transition: opacity 250ms,visibility 250ms;
  transition: opacity 250ms,visibility 250ms;
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.enter {
  opacity: 0;
  visibility: hidden;
}

.emotion-3.fade-Identifier.enter-active {
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.exit {
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.exit-active {
  opacity: 0;
  visibility: hidden;
}

.emotion-1 {
  all: initial;
  box-sizing: border-box;
  line-height: 1;
  border-radius: 2em;
  color: #333;
  background-color: yellow;
  border-color: #caca00;
  white-space: nowrap;
  min-width: 2em;
  max-width: initial;
  min-height: auto;
  max-height: initial;
  height: 2em;
  padding: 0.25em 0.5em;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: initial;
  overflow: visible;
  font-style: normal;
  font-weight: normal;
  font-size: 1rem;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: absolute;
  top: -1em;
  left: -1em;
  z-index: 4000;
  cursor: default;
  -webkit-transition: opacity 250ms,visibility 250ms;
  transition: opacity 250ms,visibility 250ms;
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.enter {
  opacity: 0;
  visibility: hidden;
}

.emotion-1.fade-Identifier.enter-active {
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.exit {
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.exit-active {
  opacity: 0;
  visibility: hidden;
}

<WireframeProvider
  api={
    Object {
      "getAnnotations": [Function],
      "getParentReference": [Function],
      "highlightNote": [Function],
      "isOpen": [Function],
      "onOpen": [Function],
      "register": [Function],
      "setOpen": [Function],
      "setOptions": [Function],
      "setParentReference": [Function],
      "unregister": [Function],
    }
  }
>
  <WrappedWireframeAnnotation
    api={
      Object {
        "getAnnotations": [Function],
        "getParentReference": [Function],
        "highlightNote": [Function],
        "isOpen": [Function],
        "onOpen": [Function],
        "register": [Function],
        "setOpen": [Function],
        "setOptions": [Function],
        "setParentReference": [Function],
        "unregister": [Function],
      }
    }
  >
    <Wrapper
      className="outline"
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      show={true}
    >
      <span
        className="outline emotion-7 emotion-8"
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
      >
        <Identifier
          annotation={
            Object {
              "Component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "displayName": "withWireframeAnnotation(WireframeProvider)",
                "type": [Function],
              },
              "count": 1,
              "id": 1,
              "options": Object {
                "description": <div>
                  Description.
                </div>,
                "title": <div>
                  Title
                </div>,
              },
            }
          }
          show={true}
        >
          <CSSTransition
            className="fade-Identifier"
            classNames=""
            in={true}
            mountOnEnter={true}
            timeout={250}
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="fade-Identifier"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={250}
              unmountOnExit={true}
            >
              <IdentifierContainer
                annotation={
                  Object {
                    "Component": Object {
                      "$$typeof": Symbol(react.memo),
                      "compare": null,
                      "displayName": "withWireframeAnnotation(WireframeProvider)",
                      "type": [Function],
                    },
                    "count": 1,
                    "id": 1,
                    "options": Object {
                      "description": <div>
                        Description.
                      </div>,
                      "title": <div>
                        Title
                      </div>,
                    },
                  }
                }
                className="fade-Identifier"
                data-annotation-identifier={true}
              >
                <Memo(IdentifierBase)
                  annotation={
                    Object {
                      "Component": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "displayName": "withWireframeAnnotation(WireframeProvider)",
                        "type": [Function],
                      },
                      "count": 1,
                      "id": 1,
                      "options": Object {
                        "description": <div>
                          Description.
                        </div>,
                        "title": <div>
                          Title
                        </div>,
                      },
                    }
                  }
                  className="fade-Identifier emotion-3 emotion-0"
                  data-annotation-identifier={true}
                >
                  <IdentifierContainer
                    className="fade-Identifier emotion-3 emotion-0"
                    data-annotation-identifier={true}
                  >
                    <cite
                      className="fade-Identifier emotion-0 emotion-1 emotion-2"
                      data-annotation-identifier={true}
                    >
                      1
                    </cite>
                  </IdentifierContainer>
                </Memo(IdentifierBase)>
              </IdentifierContainer>
            </Transition>
          </CSSTransition>
        </Identifier>
        <withWireframeAnnotation(WireframeProvider)
          api={
            Object {
              "getAnnotations": [Function],
              "getParentReference": [Function],
              "highlightNote": [Function],
              "isOpen": [Function],
              "onOpen": [Function],
              "register": [Function],
              "setOpen": [Function],
              "setOptions": [Function],
              "setParentReference": [Function],
              "unregister": [Function],
            }
          }
        >
          <WireframeProvider
            api={
              Object {
                "getAnnotations": [Function],
                "getParentReference": [Function],
                "highlightNote": [Function],
                "isOpen": [Function],
                "onOpen": [Function],
                "register": [Function],
                "setOpen": [Function],
                "setOptions": [Function],
                "setParentReference": [Function],
                "unregister": [Function],
              }
            }
          >
            children
          </WireframeProvider>
        </withWireframeAnnotation(WireframeProvider)>
      </span>
    </Wrapper>
  </WrappedWireframeAnnotation>
</WireframeProvider>
`;

exports[`Wireframe: withWireframeAnnotation should render annotation identifier when open 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-0 [disabled] {
  pointer-events: none;
}

.emotion-0.outlined > [data-annotation-identifier],
.emotion-0:hover > [data-annotation-identifier] {
  -webkit-transition: opacity 0ms,visibility 0ms;
  transition: opacity 0ms,visibility 0ms;
  opacity: 1;
}

<WireframeProvider
  api={
    Object {
      "getAnnotations": [Function],
      "getParentReference": [Function],
      "highlightNote": [Function],
      "isOpen": [Function],
      "onOpen": [Function],
      "register": [Function],
      "setOpen": [Function],
      "setOptions": [Function],
      "setParentReference": [Function],
      "unregister": [Function],
    }
  }
>
  <WrappedWireframeAnnotation>
    <Wrapper
      className="outline"
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      show={false}
    >
      <span
        className="outline emotion-0 emotion-1"
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
      >
        <Identifier
          annotation={
            Object {
              "Component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "displayName": "withWireframeAnnotation(Component)",
                "type": [Function],
              },
              "count": 1,
              "id": 1,
              "options": Object {
                "description": <div>
                  Description.
                </div>,
                "title": <div>
                  Title
                </div>,
              },
            }
          }
          show={false}
        >
          <CSSTransition
            className="fade-Identifier"
            classNames=""
            in={false}
            mountOnEnter={true}
            timeout={250}
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="fade-Identifier"
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={250}
              unmountOnExit={true}
            />
          </CSSTransition>
        </Identifier>
        <withWireframeAnnotation(Component)>
          <Component>
            <div>
              MOCK COMPONENT
            </div>
          </Component>
        </withWireframeAnnotation(Component)>
      </span>
    </Wrapper>
  </WrappedWireframeAnnotation>
</WireframeProvider>
`;

exports[`Wireframe: withWireframeAnnotation should render without identifier when closed 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-0 [disabled] {
  pointer-events: none;
}

.emotion-0.outlined > [data-annotation-identifier],
.emotion-0:hover > [data-annotation-identifier] {
  -webkit-transition: opacity 0ms,visibility 0ms;
  transition: opacity 0ms,visibility 0ms;
  opacity: 1;
}

<WireframeProvider
  api={
    Object {
      "getAnnotations": [Function],
      "getParentReference": [Function],
      "highlightNote": [Function],
      "isOpen": [Function],
      "onOpen": [Function],
      "register": [Function],
      "setOpen": [Function],
      "setOptions": [Function],
      "setParentReference": [Function],
      "unregister": [Function],
    }
  }
>
  <WrappedWireframeAnnotation>
    <Wrapper
      className="outline"
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      show={false}
    >
      <span
        className="outline emotion-0 emotion-1"
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
      >
        <Identifier
          annotation={
            Object {
              "Component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "displayName": "withWireframeAnnotation(Component)",
                "type": [Function],
              },
              "count": 1,
              "id": 1,
              "options": Object {
                "description": <div>
                  Description.
                </div>,
                "title": <div>
                  Title
                </div>,
              },
            }
          }
          show={false}
        >
          <CSSTransition
            className="fade-Identifier"
            classNames=""
            in={false}
            mountOnEnter={true}
            timeout={250}
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="fade-Identifier"
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={250}
              unmountOnExit={true}
            />
          </CSSTransition>
        </Identifier>
        <withWireframeAnnotation(Component)>
          <Component>
            <div>
              MOCK COMPONENT
            </div>
          </Component>
        </withWireframeAnnotation(Component)>
      </span>
    </Wrapper>
  </WrappedWireframeAnnotation>
</WireframeProvider>
`;

exports[`Wireframe: withWireframeAnnotation should show annotation highlighted 1`] = `
.emotion-7 {
  position: relative;
}

.emotion-7 [disabled] {
  pointer-events: none;
}

.emotion-7.outlined,
.emotion-7.outline:hover {
  z-index: 5000;
}

.emotion-7.outlined > *,
.emotion-7.outline:hover > * {
  box-shadow: 0 0 0 1px #4086f7 !important;
}

.emotion-7.outlined > [data-annotation-identifier],
.emotion-7:hover > [data-annotation-identifier] {
  -webkit-transition: opacity 0ms,visibility 0ms;
  transition: opacity 0ms,visibility 0ms;
  opacity: 1;
}

.emotion-3 {
  position: absolute;
  top: -1em;
  left: -1em;
  z-index: 4000;
  cursor: default;
  -webkit-transition: opacity 250ms,visibility 250ms;
  transition: opacity 250ms,visibility 250ms;
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.enter {
  opacity: 0;
  visibility: hidden;
}

.emotion-3.fade-Identifier.enter-active {
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.exit {
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.exit-active {
  opacity: 0;
  visibility: hidden;
}

.emotion-1 {
  all: initial;
  box-sizing: border-box;
  line-height: 1;
  border-radius: 2em;
  color: #333;
  background-color: yellow;
  border-color: #caca00;
  white-space: nowrap;
  min-width: 2em;
  max-width: initial;
  min-height: auto;
  max-height: initial;
  height: 2em;
  padding: 0.25em 0.5em;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: initial;
  overflow: visible;
  font-style: normal;
  font-weight: normal;
  font-size: 1rem;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: absolute;
  top: -1em;
  left: -1em;
  z-index: 4000;
  cursor: default;
  -webkit-transition: opacity 250ms,visibility 250ms;
  transition: opacity 250ms,visibility 250ms;
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.enter {
  opacity: 0;
  visibility: hidden;
}

.emotion-1.fade-Identifier.enter-active {
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.exit {
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.exit-active {
  opacity: 0;
  visibility: hidden;
}

<WireframeProvider
  api={
    Object {
      "getAnnotations": [Function],
      "getParentReference": [Function],
      "highlightNote": [Function],
      "isOpen": [Function],
      "onOpen": [Function],
      "register": [Function],
      "setOpen": [Function],
      "setOptions": [Function],
      "setParentReference": [Function],
      "unregister": [Function],
    }
  }
>
  <WrappedWireframeAnnotation
    isHighlighted={true}
  >
    <Wrapper
      className="outline outlined"
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      show={true}
    >
      <span
        className="outline outlined emotion-7 emotion-8"
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
      >
        <Identifier
          annotation={
            Object {
              "Component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "displayName": "withWireframeAnnotation(Component)",
                "type": [Function],
              },
              "count": 1,
              "id": 1,
              "options": Object {
                "description": <div>
                  Description.
                </div>,
                "title": <div>
                  Title
                </div>,
              },
            }
          }
          show={true}
        >
          <CSSTransition
            className="fade-Identifier"
            classNames=""
            in={true}
            mountOnEnter={true}
            timeout={250}
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="fade-Identifier"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={250}
              unmountOnExit={true}
            >
              <IdentifierContainer
                annotation={
                  Object {
                    "Component": Object {
                      "$$typeof": Symbol(react.memo),
                      "compare": null,
                      "displayName": "withWireframeAnnotation(Component)",
                      "type": [Function],
                    },
                    "count": 1,
                    "id": 1,
                    "options": Object {
                      "description": <div>
                        Description.
                      </div>,
                      "title": <div>
                        Title
                      </div>,
                    },
                  }
                }
                className="fade-Identifier"
                data-annotation-identifier={true}
              >
                <Memo(IdentifierBase)
                  annotation={
                    Object {
                      "Component": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "displayName": "withWireframeAnnotation(Component)",
                        "type": [Function],
                      },
                      "count": 1,
                      "id": 1,
                      "options": Object {
                        "description": <div>
                          Description.
                        </div>,
                        "title": <div>
                          Title
                        </div>,
                      },
                    }
                  }
                  className="fade-Identifier emotion-3 emotion-0"
                  data-annotation-identifier={true}
                >
                  <IdentifierContainer
                    className="fade-Identifier emotion-3 emotion-0"
                    data-annotation-identifier={true}
                  >
                    <cite
                      className="fade-Identifier emotion-0 emotion-1 emotion-2"
                      data-annotation-identifier={true}
                    >
                      1
                    </cite>
                  </IdentifierContainer>
                </Memo(IdentifierBase)>
              </IdentifierContainer>
            </Transition>
          </CSSTransition>
        </Identifier>
        <withWireframeAnnotation(Component)>
          <Component>
            <div>
              MOCK COMPONENT
            </div>
          </Component>
        </withWireframeAnnotation(Component)>
      </span>
    </Wrapper>
  </WrappedWireframeAnnotation>
</WireframeProvider>
`;
